modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/inbound/ContentChunkHandler.java

```diff
@@ -210,4 +227,41 @@ private void resizeContentChunks(List<AndesMessage> messageList, long sequence)
         }
         return chunkList;
     }
+
+    /**
+     * Compress the content of the message and resize content chunks of the compressed message. Resized chunks
+     * will have a maximum length of maxChunkSize
+     *
+     * @param partList      Original content chunk list
+     * @param contentLength Content length of the message
+     * @return List of resized content chunks of the compressed message
+     */
+    List<AndesMessagePart> compressAndResizeChunks(List<AndesMessagePart> partList, int contentLength) {
+        byte[] compressedMessage;
+        int originalContentLength = contentLength;
+        int compressedMessageLength;
+
+        if (originalContentLength > lz4Factory.getContentCompressionThreshold()) {
+            // Compress message
+            compressedMessage = lz4Factory.getCompressedMessage(partList, originalContentLength);
```

Comment:
As discussed, can move this if block to lz4Factory for readability and encapsulation. (need to avoid exposing byte array processing at this layer.)